torch/nn/parallel/distributed.py

```diff
@@ -103,6 +103,7 @@ def _dump_DDP_relevant_env_vars():
 class _DDPUnevenInputsConfig(NamedTuple):
     ddp_join_enabled: bool
     ddp_join_divide_by_initial_world_size: bool
+    ddp_join_throw_on_early_termination: bool
```

Comment:
Nit: May be worth noting the purpose of this field, such as when should a user flip it.